src/app/components/navigation/ErrorClearingRouter.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { ReactNode, useEffect, useState } from "react";
+import { useLocation } from "react-router-dom";
+import { useDispatch } from "react-redux";
+import { clearError } from "../../state"
+
+interface ErrorClearingRouterProps {
+    children: ReactNode;
+  }
+
+// This component clears the error state when the location changes (i.e. when the user navigates to a new page)
+
+const ErrorClearingRouter: React.FC<ErrorClearingRouterProps> = ({ children }) => {
```

Comment:
Question: does this component have to be a wrapper? You're not really using the `children` prop and you've already set `location` and `previousLocation` as the dependencies of `useEffect` (so it should trigger when these change), so does this component work if you remove the `children` prop and make this a standalone component that does not need to wrap other components?

Reply:
Yes seems to work fine as a standalone component so changed that now. I had thought to do it as a useEffect in IsaacApp but I believe it needs to be inside the Router.